src/main/java/service/ReportServiceImp.java

```diff
@@ -0,0 +1,37 @@
+package service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class ReportServiceImp implements ReportService {
+    private static final String MINUS_ACTIVITIES_CODES = "p";
+
+    @Override
+    public List<String> makeBalanceReport(List<String> activitiesList) {
+        int sum;
+        List<String> result = new ArrayList<>();
+        result.add("fruit,quantity");
+
+        Map<String, List<String[]>> stringListMap = activitiesList.stream()
+                .map(line -> line.split(","))
+                .collect(Collectors.groupingBy(strings -> strings[1]));
+
+        for (String fruit:
+                stringListMap.keySet()) {
+            sum = 0;
+            for (String[] activity:
+                    stringListMap.get(fruit)) {
+                sum += checkType(activity);
+            }
+            result.add(fruit + "," + sum);
+        }
+        return result;
+    }
+
+    private int checkType(String[] activity) {

```

Comment:
this solution will be hard to scale if we will extend types of operation. Instead create handler for each operation and put it in the map where key is operation symbol and value is the object of corresponding implementation. Then you will just call `Handler currentHandler handlers.get(operationLetter);`
and apply it `currentHandler.apply(activity)`. Basically it's nice to use strategy pattern here